### PR TITLE
Support empty invert_mask in measument gate deserialization

### PR DESCRIPTION
**Pull request by NoureldinYosri**

### CODE REVIEW

**Comment by dstrain115** [comment]
Can you explain what legacy behavior this is restoring?  I am not sure I understand what got broken.

**Comment by NoureldinYosri** [comment]
@dstrain115 An empty list `arg` is assumed to be a list of boolean

This breaks measurement gate deserialization when the invertion mask is empty because during serialization it's not set. so during deserialization the value becomes `None` instead of an empty list.

File: cirq-google/cirq_google/serialization/circuit_serializer.py [hunk 1]
```diff
@@ -546,18 +546,18 @@ def _deserialize_gate_op(
...
                 required_arg_name=None,
             )
-            invert_mask = cast(
-                List[bool],
-                arg_func_langs.arg_from_proto(
-                    operation_proto.measurementgate.invert_mask,
-                    arg_function_language=arg_function_language,
-                    required_arg_name=None,
-                ),
+            invert_mask_ = arg_func_langs.arg_from_proto(
```
Comment (dstrain115) [inline cirq-google/cirq_google/serialization/circuit_serializer.py:549]: Having `invert_mask` and `invert_mask_` is pretty confusing.  Can we give one a better name?

File: cirq-google/cirq_google/serialization/circuit_serializer_test.py [hunk 2]
```diff
@@ -660,3 +660,11 @@ def test_no_constants_table():
 
     with pytest.raises(ValueError, match='Proto has references to constants table'):
         serializer._deserialize_gate_op(op)
+
+
+def test_deserialize() -> None:
```
[the remark this answers is not in the document]
Comment (NoureldinYosri) [reply cirq-google/cirq_google/serialization/circuit_serializer_test.py]: done

File: cirq-google/cirq_google/serialization/circuit_serializer.py (hunk 1 above)
Comment (NoureldinYosri) [reply cirq-google/cirq_google/serialization/circuit_serializer.py]: renamed to `parsed_invert_mask`

**Review by dstrain115 — APPROVED** [review]

**Pull request merged**